lib/streamlit/elements/button.py

```diff
@@ -408,11 +408,16 @@ def marshall_file(
     else:
         raise RuntimeError("Invalid binary data format: %s" % type(data))
 
-    file_url = media_file_manager.add(
-        data_as_bytes,
-        mimetype,
-        coordinates,
-        file_name=file_name,
-        is_for_static_download=True,
-    )
+    if streamlit._is_running_with_streamlit:
+        file_url = get_media_file_manager().add(
+            data_as_bytes,
+            mimetype,
+            coordinates,
+            file_name=file_name,
+            is_for_static_download=True,
+        )
+    else:
+        # When running in "raw mode", we can't access the MediaFileManager.
```

Comment:
For my edification: what is "raw" mode? And why didn't Streamlit have raw mode before?

Reply:
"Raw mode" is when you run a streamlit app script outside of streamlit:

```shell
$ python my_streamlit_app.py
```

Our policy is that Streamlit APIs should not raise exceptions when called this way (they should basically just be no-ops). We didn't need the else clause for it before because `MediaFileManager` was a singleton that was instantiated automatically at startup. But with this new pluggable `MediaFileStorage` interface, we can't auto-instantiate the singleton, and so the MediaFileManager instance won't exist if a streamlit Runtime hasn't been created. (When running in "raw mode", the Runtime will never be instantiated, so MediaFileManager will always be null.)